### PR TITLE
release-23.1: changefeedccl: add new table bytes metric for usage data

### PR DESCRIPTION
**Pull request by asg0451**
Backport 1/1 commits from #123145.

/cc @cockroachdb/release

---

Add a new metric to facilitate the new changefeed billing strategy - `changefeed.usage.table_bytes` - This is a sum of the bytes watched per changefeed. It intentionally counts tables watched by multiple changefeeds multiple times. Also add auxiliary metrics about usage data collection: `changefeed.usage.error_count` (counter of errors encountered running the query that generates the table_bytes metric) and `changefeed.usage.query_duration` (histogram of how long the latter took). The collection of this data and emission of these metrics are gated behind an environment variable only intended to be set in ccloud: `COCKROACH_ENABLE_CLOUD_BILLING_ACCOUNTING`

Epic: CRDB-37031
Resolves: #121781

Release note (enterprise change): Add a new metric to facilitate the new changefeed billing strategy for usage in Cockroach Cloud.

Release justification: This is a necessary change for upcoming billing changes.

### CODE REVIEW

File: pkg/sql/plpgsql/parser/BUILD.bazel [hunk 1]
```diff
@@ -36,6 +36,7 @@ go_library(
     importpath = "github.com/cockroachdb/cockroach/pkg/sql/plpgsql/parser",
     visibility = ["//visibility:public"],
     deps = [
+        "//pkg/build",
```
Comment (asg0451) [inline pkg/sql/plpgsql/parser/BUILD.bazel:39]: idk why it generated this

File: pkg/settings/int.go [hunk 2]
```diff
@@ -178,3 +178,17 @@ func NonNegativeIntWithMaximum(maxValue int64) func(int64) error {
 		return nil
 	}
 }
+
+// NonNegativeIntWithMinimum returns a validation function that can be
+// passed to RegisterIntSetting.
+func NonNegativeIntWithMinimum(minValue int64) func(int64) error {
```
Comment (asg0451) [inline pkg/settings/int.go:184]: added this to support the validations i wanted with the old style of settings

**Comment by rharding6373** [comment]
Could you fix the build failure?

```
pkg/ccl/changefeedccl/changefeed_processors.go:1205:19: undefined: timeutil.RunWithTimeout
pkg/ccl/changefeedccl/metrics.go:354:4: unknown field 'BucketConfig' in struct literal of type metric.HistogramOptions
```

**Comment by asg0451** [comment]
@rharding6373 fixed

**Review by rgcase — APPROVED** [review]

**Review by rharding6373 — APPROVED** [review]

**Review by rharding6373 — APPROVED** [review]
:lgtm: Pending the minor comments below. Thanks for getting this backport to the finish line!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/125528#-:-O-qyZtI-XNqwWiCpnl9:b-oosuxv)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @asg0451, @nameisbhaskar, @vidit-bhat, @wenyihu6, and @xinhaoz)

___
*[`pkg/ccl/changefeedccl/metrics.go` line 356 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/125528#-O-qxue26JiiuqdsoFWf:-O-qxue26JiiuqdsoFWg:b-b1xio) ([raw file](https://github.com/cockroachdb/cockroach/blob/d4d2e81656f3e5256994397aa0e80782944e8c9a/pkg/ccl/changefeedccl/metrics.go#L356)):*
> ```Go
> 			Duration: histogramWindow,
> 			// AKA BucketConfig: metric.LongRunning60mLatencyBuckets.
> 			Buckets: prometheus.ExponentialBucketsRange(500e6, 2600e9,
> ```

Could you provide a bit more context into what these numbers mean in the comment please?
___
*[`pkg/ccl/changefeedccl/fetch_table_bytes_test.go` line 190 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/125528#-O-qy1X-DqPj0u-1AW3e:-O-qy1X-DqPj0u-1AW3f:b-z48ihj) ([raw file](https://github.com/cockroachdb/cockroach/blob/d4d2e81656f3e5256994397aa0e80782944e8c9a/pkg/ccl/changefeedccl/fetch_table_bytes_test.go#L190)):*
> ```Go
> 	res.Scan(&feedJobId)
> 
> 	// TODO: make trackers curl the other process' prom endpoint and grep etc
> ```

Is this TODO still relevant? Can this be removed?
___
*[`pkg/ccl/changefeedccl/fetch_table_bytes_test.go` line 499 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/125528#-O-qy9DkEKmPjtUXQFl6:-O-qy9DkEKmPjtUXQFl7:b-rwaq8j) ([raw file](https://github.com/cockroachdb/cockroach/blob/d4d2e81656f3e5256994397aa0e80782944e8c9a/pkg/ccl/changefeedccl/fetch_table_bytes_test.go#L499)):*
> ```Go
> 		return err
> 	}
> 	// if tenant...
> ```

Not sure what this comment means. Could you expand on it for future code readers or remove it if it's no longer necessary?

**Pull request merged**